qiskit/pulse/builder.py

```diff
@@ -677,6 +677,9 @@ def get_context(self) -> ScheduleBlock:
     @_requires_backend
     def num_qubits(self):
         """Get the number of qubits in the backend."""
+        # backendV2
+        if hasattr(self.backend, "target"):
```

Comment:
Is there any reason not to use `isinstance(self.backend, BackendV2)`, maybe circular import? If required information is stored in the backend.target then it looks reasonable to use this condition. Because we can dynamically add any attribute to the instance, it could have `.target` that may provide unexpected value. I think instance check is much safer in this sense.

Reply:
Fixed at https://github.com/Qiskit/qiskit-terra/pull/10150/commits/a21cc84d657c3cc5b9d1d6a8b0d22c733c8bae32.